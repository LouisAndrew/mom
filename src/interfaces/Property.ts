import { FluidObject } from 'gatsby-image';

export type Property = {
    name: string;
    address: string;
    propertyType: 'house' | 'apartment' | 'kavling' | 'home-office';
    saleType: 'sell' | 'rent';

    // optional fields..
    location?: string;
    imgs?: { image: string | FluidObject | FluidObject[]; imgAlt: string }[];
    description?: string;
    isTrending?: boolean;
    geolocation?: { lat: number; lng: number };
    price?: number;
    area?: number;
    roomQty?: number;
    bathQty?: number;
};
