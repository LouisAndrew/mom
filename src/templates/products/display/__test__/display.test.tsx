import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { Property } from 'interfaces/Property';
import Display from '..';

describe('Display component on products', () => {
    const mockProperty: Property[] = [
        {
            name: 'Rumah 1',
            address: 'Jalan rumah 1',
            propertyType: 'house',
            saleType: 'sell',

            imgs: [
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/webaliser-_TPTXZd9mOo-unsplash_pqtm1r.jpg',
                    imgAlt: 'Beautiful house',
                },
            ],
            description: 'Deskripsi bum bum bum',
            location: 'Penang',
            price: 200,
            area: 125,
        },
        {
            name: 'Rumah 2',
            address: 'Jalan apartment 2',
            propertyType: 'apartment',
            saleType: 'rent',

            imgs: [
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/jacques-bopp-Hh18POSx5qk-unsplash_lweaht.jpg',
                    imgAlt: 'Beautiful house 2',
                },
            ],
            description: 'Deskripsi bum bum bum',
            location: 'PIK',
            price: 500,
            area: 250,
        },
        {
            name: 'Rumah 10',
            address: 'Jalan ke dufan',
            propertyType: 'home-office',
            saleType: 'rent',

            imgs: [
                {
                    image:
                        'https://res.cloudinary.com/dsvdffre0/image/upload/v1597137792/digital-marketing-agency-ntwrk-g39p1kDjvSY-unsplash_kh6gti.jpg',
                    imgAlt: 'Beautiful house 3',
                },
            ],
            description: 'Deskripsi bum bum bum',
            location: 'jkarta Timur',
            price: 700,
            area: 500,
        },
    ];

    const mockHandleSelectLocs = jest.fn(() => {
        return;
    });
    const mockHandleSelectProps = jest.fn(() => {
        return;
    });
    const mockHandleSelectSale = jest.fn(() => {
        return;
    });

    const mockApplyFilters = jest.fn(() => {
        return;
    });

    const el: React.ReactElement = (
        <Display
            display={mockProperty}
            blur={true}
            handleSelectLocations={mockHandleSelectLocs}
            handleSelectPropertyType={mockHandleSelectProps}
            handleSelectSaleType={mockHandleSelectSale}
            applyFilters={mockApplyFilters}
        />
    );

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(el, div);
    });

    it('Should render children cards correctly', () => {
        const { queryByText } = render(el);
        mockProperty.forEach(o => {
            const textEl: HTMLElement | null = queryByText(o.name);
            if (textEl) {
                expect(textEl).toBeInTheDocument();
            } else fail();
        });
    });

    // TODO: why this test fails?
    // it('Should render with tags and calls its corresponding function correctly', () => {
    //     const { queryAllByText } = render(el);
    //     mockProperty.forEach(o => {
    //         const locsEl: HTMLElement[] = queryAllByText(
    //             o.location ? o.location : ''
    //         );
    //         const propsEl: HTMLElement[] = queryAllByText(o.propertyType);
    //         const saleEl: HTMLElement[] = queryAllByText(o.saleType);

    //         if (locsEl.length > 0) {
    //             fireEvent.click(locsEl[0]);
    //             expect(mockHandleSelectLocs).toBeCalled();
    //         }

    //         if (propsEl.length > 0) {
    //             fireEvent.click(propsEl[0]);
    //             expect(mockHandleSelectProps).toBeCalled();
    //         } else fail();

    //         if (saleEl.length > 0) {
    //             fireEvent.click(saleEl[0]);
    //             expect(mockHandleSelectSale).toBeCalled();
    //         }
    //     });
    // });

    it('matches snapshot', () => {
        const tree = renderer.create(el).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
