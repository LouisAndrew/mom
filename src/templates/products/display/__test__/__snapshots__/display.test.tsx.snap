// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component on products matches snapshot 1`] = `
<div
  className="sc-fzoNJl kIBoOM"
  display="flex"
  width={1}
>
  <div
    className="sc-AxmLO bOahTC"
    display="flex"
    overflow="hidden"
    width={
      Array [
        0.9,
        0.8,
        0.45,
        0.4,
      ]
    }
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="sc-fzpans cWuRwg"
        height={
          Array [
            "auto",
          ]
        }
        overflow="hidden"
        width={
          Array [
            1,
          ]
        }
      >
        <img
          alt="Beautiful house"
          src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/webaliser-_TPTXZd9mOo-unsplash_pqtm1r.jpg"
        />
      </div>
      <div
        className="sc-fzozJi deZYli"
        display="flex"
        width={1}
      >
        <h3
          className="sc-AxiKw dsegVB"
          fontFamily="heading"
          fontSize={
            Array [
              3,
              3,
              4,
              4,
            ]
          }
        >
          Rumah 1
        </h3>
        <p
          className="sc-AxhUy iPnAqk"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
        >
          Jalan rumah 1
        </p>
        <div
          className="sc-fzoLsD kMKfsv"
          display="flex"
          width={1}
        >
          <span
            className="sc-Axmtr iNVGjd"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            Penang
          </span>
          <span
            className="sc-Axmtr ibIjVG"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            Rumah
          </span>
          <span
            className="sc-Axmtr jcmgys"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={2}
          >
            Jual
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-fzozJi bngyJH"
      width={1}
    >
      <h2
        className="sc-AxirZ jrdOBY"
        fontFamily="heading"
        fontSize={
          Array [
            4,
            4,
            5,
            5,
          ]
        }
      >
        200 Jt
      </h2>
      <button
        className="sc-AxheI eICHae"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        onClick={[Function]}
        width={
          Array [
            1,
          ]
        }
      >
        More info
      </button>
    </div>
  </div>
  <div
    className="sc-AxmLO bOahTC"
    display="flex"
    overflow="hidden"
    width={
      Array [
        0.9,
        0.8,
        0.45,
        0.4,
      ]
    }
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="sc-fzpans cWuRwg"
        height={
          Array [
            "auto",
          ]
        }
        overflow="hidden"
        width={
          Array [
            1,
          ]
        }
      >
        <img
          alt="Beautiful house 2"
          src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137791/jacques-bopp-Hh18POSx5qk-unsplash_lweaht.jpg"
        />
      </div>
      <div
        className="sc-fzozJi deZYli"
        display="flex"
        width={1}
      >
        <h3
          className="sc-AxiKw dsegVB"
          fontFamily="heading"
          fontSize={
            Array [
              3,
              3,
              4,
              4,
            ]
          }
        >
          Rumah 2
        </h3>
        <p
          className="sc-AxhUy iPnAqk"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
        >
          Jalan apartment 2
        </p>
        <div
          className="sc-fzoLsD kMKfsv"
          display="flex"
          width={1}
        >
          <span
            className="sc-Axmtr iNVGjd"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            PIK
          </span>
          <span
            className="sc-Axmtr fEcKHw"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            Apartment
          </span>
          <span
            className="sc-Axmtr hFDSlT"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={2}
          >
            Sewa
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-fzozJi bngyJH"
      width={1}
    >
      <h2
        className="sc-AxirZ jrdOBY"
        fontFamily="heading"
        fontSize={
          Array [
            4,
            4,
            5,
            5,
          ]
        }
      >
        500 Jt
      </h2>
      <button
        className="sc-AxheI eICHae"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        onClick={[Function]}
        width={
          Array [
            1,
          ]
        }
      >
        More info
      </button>
    </div>
  </div>
  <div
    className="sc-AxmLO bOahTC"
    display="flex"
    overflow="hidden"
    width={
      Array [
        0.9,
        0.8,
        0.45,
        0.4,
      ]
    }
  >
    <div
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <div
        className="sc-fzpans cWuRwg"
        height={
          Array [
            "auto",
          ]
        }
        overflow="hidden"
        width={
          Array [
            1,
          ]
        }
      >
        <img
          alt="Beautiful house 3"
          src="https://res.cloudinary.com/dsvdffre0/image/upload/v1597137792/digital-marketing-agency-ntwrk-g39p1kDjvSY-unsplash_kh6gti.jpg"
        />
      </div>
      <div
        className="sc-fzozJi deZYli"
        display="flex"
        width={1}
      >
        <h3
          className="sc-AxiKw dsegVB"
          fontFamily="heading"
          fontSize={
            Array [
              3,
              3,
              4,
              4,
            ]
          }
        >
          Rumah 10
        </h3>
        <p
          className="sc-AxhUy iPnAqk"
          fontFamily="body"
          fontSize={
            Array [
              1,
              1,
              2,
              2,
            ]
          }
        >
          Jalan ke dufan
        </p>
        <div
          className="sc-fzoLsD kMKfsv"
          display="flex"
          width={1}
        >
          <span
            className="sc-Axmtr iNVGjd"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            jkarta Timur
          </span>
          <span
            className="sc-Axmtr gieWuX"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={1}
          >
            Ruko
          </span>
          <span
            className="sc-Axmtr hFDSlT"
            color="dark.3"
            fontFamily="body"
            fontSize={
              Array [
                1,
                1,
                2,
                2,
              ]
            }
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={2}
          >
            Sewa
          </span>
        </div>
      </div>
    </div>
    <div
      className="sc-fzozJi bngyJH"
      width={1}
    >
      <h2
        className="sc-AxirZ jrdOBY"
        fontFamily="heading"
        fontSize={
          Array [
            4,
            4,
            5,
            5,
          ]
        }
      >
        700 Jt
      </h2>
      <button
        className="sc-AxheI eICHae"
        fontFamily="body"
        fontSize={
          Array [
            1,
            1,
            2,
            2,
          ]
        }
        fontWeight="bold"
        onClick={[Function]}
        width={
          Array [
            1,
          ]
        }
      >
        More info
      </button>
    </div>
  </div>
</div>
`;
